lib/publiccloud/ec2.pm

```diff
@@ -23,28 +23,21 @@ sub create_credentials {
     my ($self) = @_;
 
     if (!defined($self->credentials)) {
-        my $credentials = $self->do_rest_api('/ec2/users', method => 'post');
-        die("Missing key") if (!defined($credentials->{keys}));
-        for my $key (@{$credentials->{keys}}) {
-            next if (!defined($key->{secret}));
-            next if ($key->{status} ne 'active');
-            $self->key_id($key->{key_id});
-            $self->key_secret($key->{secret});
-        }
+        my $credentials = $self->do_rest_api('/ec2/key', method => 'post');
         $self->credentials($credentials);
-        die('Failed to create credentials') unless (defined($self->key_id) && defined($self->key_secret));
+        $self->key_id($credentials->{key_id});
```

Comment:
`$credentials` contains `$kei_id` and `$secret` , and we have additionally `$key_id` and`$key_secret`. Any chance to simplify it ?

Reply:
I do not wonne remove the possibility to set the credentials as constructor parameter. And I need something to know, if the key was "auto-generated" or not. So from my point not.